File: opbit.c
#include <stdio.h>
#include <math.h>
int main()
{
	int t;
	scanf("%d",&t);
	while(t--)
	{
		int a,b,a1,b1,nd,xor,i;
		scanf("%d%d",&a,&b);
		a1=(int)sqrt((float)a);
		b1=(int)sqrt((float)b);
		nd=(a1+1)*(a1+1)-a;
		xor=nd;
		for(i=a1+1;i<b1;i++)
		{
			int j=(i+1)*(i+1)-i*i;
			nd=nd&j;
			xor=xor^j;
		}
		printf("%d\n", nd&xor);
	}
	return 0;
}

File: opbit.py
import math
t=input()
while t:
	t-=1
	a,b=raw_input().split()
	a,b=int(a),int(b)
	ans=0
	if a!=b:
		a1,b1=math.sqrt(a),math.sqrt(b)
		a1,b1=int(a1),int(b1)
		xor=(a1+1)**2-a;nd=xor
		for i in range(a1+1,b1):
			i=(i+1)**2-i**2
			xor=xor^i
			nd=nd&i
		ans=xor&nd
	print ans